### PR TITLE
Consolidated and updated AAD token access methods

### PR DESCRIPTION
**Pull request by maheshkelkar**
Also,
- using null tracer (disable zipkin)
- Use default port if no port specified in the URL

### CODE REVIEW

**Comment by PeterFCaswell** [comment]
Looks fine to me. @rikesh-chouhan can you also look through it?

File: auth/src/main/scala/com/lookout/borderpatrol/auth/tokenmaster/OAuth2.scala [hunk 1]
```diff
@@ -144,7 +144,7 @@ object OAuth2 {
      * @return
      */
     def codeToClaimsSet(req: BorderRequest, loginManager: OAuth2LoginManagerMixin):
-      Future[(JWTClaimsSet, JWTClaimsSet)] = {
+      Future[(String, JWTClaimsSet, JWTClaimsSet)] = {
```
Comment (rikesh-chouhan) [inline auth/src/main/scala/com/lookout/borderpatrol/auth/tokenmaster/OAuth2.scala]: Can we make this a class? The usage of _2 and _1 to access values makes it little ugly.

**Comment by rikesh-chouhan** [comment]
Looks good to me except for 1 portion that I commented on

File: auth/src/main/scala/com/lookout/borderpatrol/auth/tokenmaster/OAuth2.scala (hunk 1 above)
Comment (maheshkelkar) [inline auth/src/main/scala/com/lookout/borderpatrol/auth/tokenmaster/OAuth2.scala]: We can, but then we will be ignoring Scala matching capabilities. I mean
`val (a, b) = codeClaimsSet(...)` IMO is better than

As you can see, we are not really `._1` in the code. Thats mainly in the test code, and I can get rid of that one.

**Pull request merged**